### PR TITLE
quota: detect/error if cpu-set is used with cgroup v1

### PR DESCRIPTION
**Pull request by mvo5**
This commit adds a check that cpu-set resources can only be set
on systems with cgroup v2.

This is a followup for https://github.com/snapcore/snapd/pull/11534#discussion_r834116997 - the cgroup v2 check was only done client side but must be done on the backend. 

Setting to draft as I'm not 100% sure this is the right place, the `servicestate/quota_control.go` would also be an option but it does not have a natural place to validate a detail like this (AFAICT). Plus we already have systemd handling there so cgroup detection seems not too out of place.

### CODE REVIEW

File: snap/quota/resources.go [hunk 1]
```diff
@@ -85,6 +95,13 @@ func (qr *Resources) validateCPUSetQuota() error {
 	if len(qr.CPUSet.CPUs) == 0 {
 		return fmt.Errorf("cpu-set quota must not be empty")
 	}
+	if cgroupVerErr != nil {
+		return cgroupVerErr
+	}
+	if cgroupVer < 2 {
```
Comment (pedronis) [inline snap/quota/resources.go:101]: this is a convenient place for this but afaict this code is also called by Group.validate which is used also by ResolveCrossReferences, and that's a strange place to do this check from.  

I think what would be better is to have a separate Resources.CheckFeatureRequiremens() error method do this kind of checks, and call it from  UpdateQuotaLimits after the call to validate.

**Review by pedronis — CHANGES_REQUESTED** [review]

File: snap/quota/resources.go (hunk 1 above)
Comment (mvo5) [reply snap/quota/resources.go]: Thanks, I added this method and call it in servicestate for CreateQuota and UpdateQuota now.

File: overlord/servicestate/quota_control.go [hunk 2]
```diff
@@ -98,6 +102,10 @@ func CreateQuota(st *state.State, name string, parentName string, snaps []string
 	if err := resourceLimits.Validate(); err != nil {
 		return nil, fmt.Errorf("cannot create quota group %q: %v", name, err)
 	}
+	// validate that the system has the features needed for this resource
+	if err := resourcesCheckFeatureRequirements(&resourceLimits); err != nil {
```
Comment (pedronis) [inline overlord/servicestate/quota_control.go:106]: these looks like the right placea as they are also the ones calling quotaGroupsAvailable

**Review by pedronis — APPROVED** [review]
thanks

**Review by Meulengracht — APPROVED** [review]
Looks really good, thanks for taking care of this! LGTM

**Pull request merged**